storage/scylla_api.c

```diff
@@ -0,0 +1,707 @@
+  }
+  for (size_t i = 0; i < select_num; i++) {
+    statement = ret_select_from_bundleTable_statement(select_prepared, select_method, select_where + i);
+    select_from_bundleTable(session, statement, &output, &outputNum);
+    ta_log_debug("row = %d\n", outputNum);
+    for (size_t i = 0; i < outputNum; i++) {
+      ta_log_debug("%s %s %s %s %ld %ld %s %s\n", output[i].bundle, output[i].address, output[i].hash,
+                   output[i].message, output[i].value, output[i].timestamp, output[i].trunk, output[i].branch);
+    }
+    free_transaction(&output);
+  }
+
+  cass_prepared_free(select_prepared);
+  return ret;
+}
+
+static status_t get_column_from_bundleTable(CassSession* session, hash243_queue_t* res_queue,
+                                            const SelectMethod select_method, const Select_Where* select_where,
+                                            const char* column_name) {
+  status_t ret = SC_OK;
+  CassStatement* statement = NULL;
+  const CassPrepared* select_prepared = NULL;
+  if (select_where == NULL) {
+    return SC_STORAGE_INVAILD_INPUT;
+  }
+  if (prepare_query(session, select_query[select_method].query, &select_prepared) != CASS_OK) {
+    ta_log_error("%s\n", "prepare SELECT query fail");
+    return SC_STORAGE_CASSANDRA_QUREY_FAIL;
+  }
+
+  statement = ret_select_from_bundleTable_statement(select_prepared, select_method, select_where);
+  ret = push_columns_into_queue(session, statement, res_queue, column_name);
+
+  cass_prepared_free(select_prepared);
+  return ret;
+}
+status_t get_column_from_edgeTable(CassSession* session, hash243_queue_t* res_queue, cass_byte_t* edge,
+                                   const char* column_name) {
+  status_t ret = SC_OK;
+  static const char* query = "SELECT * FROM permanent.edgeTable WHERE edge = ?";
+  CassStatement* statement = NULL;
+  const CassPrepared* select_prepared = NULL;
+
+  if (prepare_query(session, query, &select_prepared) != CASS_OK) {
+    ta_log_error("%s\n", "prepare SELECT query fail");
+    return SC_STORAGE_CASSANDRA_QUREY_FAIL;
+  }
+
+  statement = cass_prepared_bind(select_prepared);
+  cass_statement_bind_bytes_by_name(statement, "edge", edge, FLEX_TRIT_SIZE_243);
+  ret = push_columns_into_queue(session, statement, res_queue, column_name);
+
+  cass_prepared_free(select_prepared);
+  return ret;
+}
+
+status_t get_transactions(CassSession* session, hash243_queue_t* res_queue, hash243_queue_t bundles,
+                          hash243_queue_t addresses, hash243_queue_t approves) {
+  status_t ret = SC_OK;
+  hash243_queue_t itr243 = NULL;
+  hash243_queue_t itr243_2 = NULL;
+  Select_Where select_where;
+
+  CDL_FOREACH(bundles, itr243) {
+    select_where.bundle = (cass_byte_t*)itr243->hash;
+    ret = get_column_from_bundleTable(session, res_queue, WITH_BUNDLE, &select_where, "hash");
+    if (ret != SC_OK) {
+      goto exit;
+    }
+  }
+  CDL_FOREACH(addresses, itr243) {
+    hash243_queue_t addr_bundles = NULL;
```

Comment:
May I ask what is addr_bundles?

Reply:
Bundle hashes in an address.
What if i modify it to "bundles" simply.

Reply:
normally we call the identifier of a bundle `bundle_hash`